app/players/player.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+
+class Player(ABC):
+    def __init__(self, nickname: str) -> None:
+        self.nickname = nickname
+
+    @abstractmethod

```

Comment:
all the code is supernice, only one suggestion - I am not sure whether annotations for abstract methods should be None or should demonstrate return type in the real realisation (non-abstr version of this method). But of course I will not reject for this. cool!

Reply:
This is actually an interesting question
Different classes inherited from an abstract class can return different things. Maybe something like "any" will be more appropriate? idk

Reply:
In this case, methods don't return anything, so None is ok